Add new adjustment history charts

samnlindsay opened the pull request:
For the time being just adds a new chart to the existing menu of charts provided and sticks them at the bottom of the `all_charts_write_html_file()` output

Still to address:
- Arrangement of charts in all charts output
- New chart for distribution of gamma levels for each col (to complete link between m/u probabilities and influence factor)
- Replace adjustment/influence factor with Bayes factor, K in charts and documentation (worthy of a separate PR)

RobinL commented:
In general this looks good but I think there are some edge cases where the charts don't come out quite right e.g. where there are low numbers of iterations.  Not sure the best solution.  Here's an example:
 
![image](https://user-images.githubusercontent.com/2608005/88207221-6526b000-cc47-11ea-8367-419240097342.png)

where you get fractional iterations on the axis and the bars are quite thing.

I was fiddling with the spec in the [vega editor](https://vega.github.io/editor/#/edited)

This is by no means complete but i tried to replicate grouped bars
https://gist.github.com/RobinL/55df885a70eca16201c2aabec8de7696

I wonder if grouped bars would be 'good enough' if we hid the x axis?  I'm assuming you didn't use them because it looks a bit rubbish?

Sorry - i've run out of time today but wanted to write this comment, mainly so i remember where i got to with this

samnlindsay commented:
Reverting to line chart for the history means no need for jitter and other hacks, so the iteration axis can be set to ordinal and everything's fine.

samnlindsay commented:
Just to clarify the changes:
- Added a line to `_convert_params_dict_to_normalised_adjustment_data` to add a column `level_proportion` to the output
- New `gamma_distribution_chart` to mirror other charts and show that proportion (with "non-null" caveat)
- Changed the colour scheme to make m (green) and u (red) probability distributions more intuitive
- Tidied up differences between two datasets used in the charts (e.g. "column" vs "col_name", "0" vs "level_0")
- Rearranged charts in `all_charts_write_html_file`

RobinL commented on splink/params.py:
```diff
@@ -196,6 +200,44 @@ def _convert_params_dict_to_normalised_adjustment_data(self):
+        """
+        adj_data = []
+
+        pi = gk = self.params["π"]
```
is the ` = gk` bit a typo here?

samnlindsay replied:
If it is, it's one of yours that I copied:
https://github.com/moj-analytical-services/splink/blob/master/splink/params.py#L181

RobinL replied:
touché, haha.  i will fix, gimme a sec

RobinL commented:
Looks good - happy with this now.  I spent quite a lot of time messing about with colour schemes without really coming up with a better solution.  I wanted to find a sort of 'red orange green' colour scheme, but there doesn't seem to be an easy way of creating a custom domain and range that works in the same way as the scale.

so e.g.:

`range: ["red", "orange", "green"]
`

will map to 
- red and orange in the case of two levels and 
- red orange green in the case of three

which probably isn't what we want.  

Only thing I wonder is whether `viridis` may be easer to see than `redyellowgreen` (although I admit, it's a bit less intuitive in terms of correspondence between colours and levels)
